### PR TITLE
Display a short message before reporting a pub site issue.

### PR DESCRIPTION
**Pull request by isoos**
You can try this in staging, by clicking on the "report an issue" link at the bottom of the page: https://dartlang-pub-dev.appspot.com/

screen shot 2018-09-06 at 17 37 09

I'm not sure what the best wording would be, /cc @mit-mit

### CODE REVIEW

**Comment by kevmoo** [comment]
CC @mit-mit – I really don't like this UX. These types of pop-ups aren't super clear about what clicking "okay" means vs cancel.

I realize it's easier to implement than pulling in orimplementing some other type of pop-up JS lib, though...

**Comment by isoos** [comment]
> I realize it's easier to implement than pulling in orimplementing some other type of pop-up JS lib, though...

Yeah, it is a tradeoff between spending a few minutes with each mis-reported issue vs. displaying something that provides enough info about the correct process (and it is easy to implement).

File: static/js/script.dart [hunk 1]
```diff
@@ -316,6 +317,18 @@ void _setEventForCheckboxChanges() {
   });
 }
 
+void _guardReportIssue() {
+  for (AnchorElement bugLink in document.querySelectorAll('a.github_issue')) {
+    bugLink.onClick.listen((event) {
+      if (!window.confirm('This link is for reporting issues for the pub site. '
+          'If you would like to report a problem with a package, please visit '
+          'its homepage or contact its developers.')) {
+        event.preventDefault();
```
Comment (sortie) [inline static/js/script.dart]: Not sure if the browser popup is the right UI, maybe a html window of our own making is better. Browser popup dialogues are something of an happening and not the nicest UI.

If we use a custom window, and we're on a package page, we could link people to the package's homepage. I'd love if we did #1522 and could take advantage of that.

The message looks fine to me. If we go with this approach, we could have a tech writer or someone review that the message is clearly understood by people in a hurry that don't understand English as well as we do. I suspect your proposed UX doesn't quite catch the kind of people that don't pay attention to which bug tracker they are on.

Comment (isoos) [reply static/js/script.dart]: > If we use a custom window, and we're on a package page, we could link people to the package's homepage. I'd love if we did #1522 and could take advantage of that.

Agreed. Shall we land this until that happens, or shall we leave it as it is now?

Comment (sortie) [reply static/js/script.dart]: I'm OK if this lands as it is, but consider if you'll want to invest some time in improving the UX a bit, or whether we'll file an issue about doing that.

Comment (isoos) [reply static/js/script.dart]: Tracking in https://github.com/dart-lang/pub-dartlang-dart/issues/1594

**Comment by mit-mit** [comment]
@isoos how often do we actually get package bugs reported here?

**Comment by isoos** [comment]
> how often do we actually get package bugs reported here?

2-5 per week, but the trend is increasing in the past few months.

**Pull request merged**